src/spec/context/procsy.cr

```diff
@@ -0,0 +1,15 @@
+module Spec
+  abstract class Context
+    struct Procsy
```

Comment:
That's one weird name... did you mean `Proxy`?

Reply:
I stole it from RSpec 😊 

https://www.rubydoc.info/github/rspec/rspec-core/RSpec/Core/Example/Procsy

It's called Procsy because it's a proxy to a Proc 😉

Reply:
It's likely a word pun for a proc proxy, that's pretty funny.